### PR TITLE
rgw: Erase old storage class attr when the object is rewrited using r…

### PR DESCRIPTION
**Pull request by xijiacun**
…adosgw-admin.

Fixes: https://tracker.ceph.com/issues/53790
Signed-off-by: zhiming zhang <zhangzhm1@chinatelecom.cn>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by xijiacun** [comment]
@smithfarm @tchaikov Please help to review the code. Thanks

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -3664,6 +3664,7 @@ int RGWRados::rewrite_obj(rgw::sal::Object* obj, const DoutPrefixProvider *dpp,
 
   attrset.erase(RGW_ATTR_ID_TAG);
   attrset.erase(RGW_ATTR_TAIL_TAG);
+  attrset.erase(RGW_ATTR_STORAGE_CLASS);
```
Comment (cbodley) [inline src/rgw/rgw_rados.cc:3667]: ok, this is makes sense i guess

ideally `object rewrite` would preserve the original storage class, and support the `--storage-class=` option to override what's there, but this looks like the simplest fix

**Review by cbodley — APPROVED** [review]

**Comment by cbodley** [comment]
passed qa https://pulpito.ceph.com/cbodley-2022-01-27_22:43:41-rgw-wip-cbodley-testing2-distro-default-smithi (rerun https://pulpito.ceph.com/cbodley-2022-01-28_15:14:53-rgw-wip-cbodley-testing2-distro-default-smithi/)

**Comment by cbodley** [comment]
@xijiacun can you please rebase your commit on top of master? the extra commit 'Merge branch 'ceph:master' into master' commit is all that's blocking merge

**Comment by xijiacun** [comment]
> @xijiacun can you please rebase your commit on top of master? the extra commit 'Merge branch 'ceph:master' into master' commit is all that's blocking merge

OK，what about now?

**Comment by xijiacun** [comment]
@cbodley

**Comment by xijiacun** [comment]
> @xijiacun it looks like another merge commit snuck in

@cbodley OK, now only one commit left, what about now?

**Comment by cbodley** [comment]
thanks @xijiacun!

**Pull request merged**